org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/proposals/NewCUUsingWizardProposal.java

```diff
@@ -109,12 +111,17 @@ public class NewCUUsingWizardProposal extends ChangeCorrectionProposal {
 	private ITypeBinding fSuperType;
 
 	private boolean fShowDialog;
+	private boolean fCallSemanticHighlightingReconciler;
 
 	public NewCUUsingWizardProposal(ICompilationUnit cu, Name node, int typeKind, IJavaElement typeContainer, int severity) {
-		this(cu, node, typeKind, typeContainer, null, severity);
+		this(cu, node, typeKind, typeContainer, null, severity, false);
 	}
 
 	public NewCUUsingWizardProposal(ICompilationUnit cu, Name node, int typeKind, IJavaElement typeContainer, ITypeBinding superType, int severity) {
+		this(cu, node, typeKind, typeContainer, superType, severity, true);
+	}
+
+	private NewCUUsingWizardProposal(ICompilationUnit cu, Name node, int typeKind, IJavaElement typeContainer, ITypeBinding superType, int severity, boolean callSemanticHighlightingReconciler) {
 		super("", null, severity, null); //$NON-NLS-1$
```

Comment:
Will the check of superType != null not sufficient to force semantic highlighting?

Reply:
If you are sure that the superType will not change because of the new type creation, then this call is not required.